### ISSUE TITLE
using spotbugs

### ISSUE THREAD

**Issue by eroux** [BuddhistDigitalResourceCenter/lds-pdi#35]
We should use tools like [spotbugs](https://spotbugs.github.io/), these tools detect memory leaks quite efficiently. Maybe [checkstyle](https://github.com/checkstyle/checkstyle) could be useful too (for a slightly different purpose)

**Comment by MarcAgate** [comment]
I have used spotbugs who didn't reveal anything special but minor points about declaring as "protected or final" some members of various classes.
Then I ran jVisualVM and YourKit (which is amazing). As far as I can tell from that, this is what I observed, hitting as much as I could my local version, even requesting 1000 results for etexts :
-memory usage remains stable over time
-all input and output streams are properly closed
-all object of lds-pdi are referenced and accessible by GC except for a couple of QueryTemplate json objects (I have to validate the idea that these unreachable objects are actually in JCS cache)
- I am going to keep tracing ldspdi objects life cycle and review caching strategy/settings. 
- I am also starting to implement a stats module for JCS cache so we can check at anytime what's going on.
